Adds new tutorial for sample chat app using Hoodie.

HipsterBrown opened the pull request:
Also adds a link to the tutorial in the tutorials menu sidebar.

Preview of PR:

![screen shot 2015-05-16 at 1 25 12 am](https://cloud.githubusercontent.com/assets/3051193/7664844/7fa27508-fb6a-11e4-8375-b3d80ec68e0b.png)

Current PR for hoodie-template-app-chat:
https://github.com/zoepage/hoodie-template-app-chat/pull/2

Open to any constructive feedback and suggestions for improving this tutorial.

HipsterBrown committed:
Adds new tutorial for sample chat app using Hoodie.

Also adds a link to the tutorial in the tutorials menu sidebar.

varjmes commented on en/tutorials/chat-app.md:
````diff
@@ -0,0 +1,400 @@
+</section>
+```
+
+Before we get started with the JavaScript, we'll add some unqiue CSS to our **www/assets/main.css** file:
````
'unqiue' should be spelt 'unique'

HipsterBrown replied:
Thanks! That one always gets me. :)

varjmes commented on en/tutorials/chat-app.md:
````diff
@@ -0,0 +1,400 @@
+}
+```
+
+This is definitely one of the more complex feature of our app but worth it to make this chat app feel more complete. As an extra challenge, try to add notifications for when the current user is @mentioned by another user in the chat by highlighting their messages with a unique background color. 
````
I think this might read better as "more complex features" rather than "more complex feature".

Hope this helps :)

HipsterBrown replied:
Great catch. That does sound a lot better now that I'm not reading it at 1 AM.

HipsterBrown committed:
Fixes spelling error and grammar correction.

janl commented on en/tutorials/chat-app.md:
````diff
@@ -0,0 +1,400 @@
+
+```js 
+// initialize Hoodie
+var hoodie  = new Hoodie();
````
funny whitespace before the `=`

janl commented on en/tutorials/chat-app.md:
```diff
@@ -0,0 +1,400 @@
+  if ( messageContent.length < 1 ) { return false; }
+
+  // create a new message model
+  var message = new messageModel(messageContent);
```
I must be missing something, where is `messageModel` coming from?

janl commented on en/tutorials/chat-app.md:
```diff
@@ -0,0 +1,400 @@
+  // using the global messageStore, add this message object and publish it to the global store.
+  messageStore.add(message).publish();
+
+  trigger an immediate sync with the server for quicker updates
```
missing comment markers: `//`

janl commented on en/tutorials/chat-app.md:
````diff
@@ -0,0 +1,400 @@
+}
+```
+
+You may have noticed the **new messageModel(messageContent)**; we define this **messageModel** just after defining the **sendMessage** function, in order to abstract some of the logic that goes into making our message object we're saving to the **messageStore**:
````
Ah, here is the `messageModel`, maybe worth making a note further up that it comes later.

janl commented:
added a few nitpicks, but overall very good quality, would love to get this merged asap!

HipsterBrown committed:
Updates chat-app.md to fix extra whitespace.

HipsterBrown committed:
Updates chat-app.md to fix certain issues

Adds missing ```//``` before ```hoodie.remote.push()``` in the ```sendMessage()``` function.

Adds comment hint about ```messageModel()``` function in ```sendMessage()``` before upcoming explanation.

HipsterBrown commented:
@janl thanks for catching those errors and great feedback. It's a good reminder of why I should re-read my tutorials the next morning after finishing them at 3 AM.

janl commented:
Thank you VEEEERY much! <333 \o/

zoepage commented:
The tutorial is really great! Thank you @HipsterBrown <3

The content needs to be added to the de folder too plus to the config so the translation works. :)

HipsterBrown commented:
@zoepage Does that need to be in a separate PR? Also, I think the to the repo is wrong at the moment.

zoepage commented:
as far as the PR is closed, just do it in a seperate PR :)
